src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Metadata/DefaultModelBindingMessageProvider.cs

```diff
@@ -0,0 +1,196 @@
+        private Func<string, string> _valueIsInvalidAccessor;
+        private Func<string, string> _valueMustBeANumberAccessor;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DefaultModelBindingMessageProvider"/> class.
+        /// </summary>
+        public DefaultModelBindingMessageProvider()
+        {
+            SetMissingBindRequiredValueAccessor(Resources.FormatModelBinding_MissingBindRequiredMember);
+            SetMissingKeyOrValueAccessor(Resources.FormatKeyValuePair_BothKeyAndValueMustBePresent);
+            SetMissingRequestBodyRequiredValueAccessor(Resources.FormatModelBinding_MissingRequestBodyRequiredMember);
+            SetValueMustNotBeNullAccessor(Resources.FormatModelBinding_NullValueNotValid);
+            SetAttemptedValueIsInvalidAccessor(Resources.FormatModelState_AttemptedValueIsInvalid);
+            SetUnknownValueIsInvalidAccessor(Resources.FormatModelState_UnknownValueIsInvalid);
+            SetValueIsInvalidAccessor(Resources.FormatHtmlGeneration_ValueIsInvalid);
+            SetValueMustBeANumberAccessor(Resources.FormatHtmlGeneration_ValueMustBeNumber);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DefaultModelBindingMessageProvider"/> class based on
+        /// <paramref name="originalProvider"/>.
+        /// </summary>
+        /// <param name="originalProvider">The <see cref="DefaultModelBindingMessageProvider"/> to duplicate.</param>
+        public DefaultModelBindingMessageProvider(DefaultModelBindingMessageProvider originalProvider)
+        {
+            if (originalProvider == null)
+            {
+                throw new ArgumentNullException(nameof(originalProvider));
+            }
+
+            SetMissingBindRequiredValueAccessor(originalProvider.MissingBindRequiredValueAccessor);
+            SetMissingKeyOrValueAccessor(originalProvider.MissingKeyOrValueAccessor);
+            SetMissingRequestBodyRequiredValueAccessor(originalProvider.MissingRequestBodyRequiredValueAccessor);
+            SetValueMustNotBeNullAccessor(originalProvider.ValueMustNotBeNullAccessor);
+            SetAttemptedValueIsInvalidAccessor(originalProvider.AttemptedValueIsInvalidAccessor);
+            SetUnknownValueIsInvalidAccessor(originalProvider.UnknownValueIsInvalidAccessor);
+            SetValueIsInvalidAccessor(originalProvider.ValueIsInvalidAccessor);
+            SetValueMustBeANumberAccessor(originalProvider.ValueMustBeANumberAccessor);
+        }
+
+        /// <inheritdoc/>
+        public override Func<string, string> MissingBindRequiredValueAccessor => _missingBindRequiredValueAccessor;
+
+        /// <summary>
+        /// Sets a value for the <see cref="MissingBindRequiredValueAccessor"/> property.
+        /// </summary>
+        /// <param name="value">The value to set.</param>
+        public void SetMissingBindRequiredValueAccessor(Func<string, string> value)
```

Comment:
Suggest using more expressive parameter names in this class e.g. `missingBindRequiredValueAccessor` for this one.